tests/integration/kubernetes/k8s-confidential-attestation.bats

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env bats
+# Copyright 2024 IBM Corporation
+# Copyright 2024 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+load "${BATS_TEST_DIRNAME}/../../common.bash"
+load "${BATS_TEST_DIRNAME}/confidential_common.sh"
+load "${BATS_TEST_DIRNAME}/tests_common.sh"
+
+export test_key="aatest"
+export SUPPORTED_TEE_HYPERVISORS=("qemu-sev" "qemu-snp" "qemu-tdx" "qemu-se")
+export SUPPORTED_NON_TEE_HYPERVISORS=("qemu")
+
+setup() {
+	if ! [[ " ${KATA_HYPERVISOR} " =~ " ${SUPPORTED_TEE_HYPERVISORS[@]} " ]] && ! [[ " ${KATA_HYPERVISOR} " =~ " ${SUPPORTED_NON_TEE_HYPERVISORS} " ]]; then
+		skip "Test not supported for ${KATA_HYPERVISOR}"
+	fi
+
+	get_pod_config_dir
+
+	# TODO need to get CC_KBS deployed with sample KBS based on info in
+	# https://github.com/confidential-containers/confidential-containers/issues/184#issuecomment-1894373294
+	# with the resource content
+	# "default/aa/key": $($test_key | base64)
+}
+
+@test "Test we can get evidence from CDH resource" {
+
+	# TODO - do we still need AA_KBC set and checked - do we have any other options?
+	if [ "${AA_KBC}" = "cc_kbc" ]; then
+		# CC KBC is specified as: cc_kbc::http://host_ip:port/, and 60000 is the default port used
+		# by the service, as well as the one configured in the Kata Containers rootfs.
+		CC_KBS_IP=${CC_KBS_IP:-"$(hostname -I | awk '{print $1}')"}
+		CC_KBS_PORT=${CC_KBS_PORT:-"60000"}
+		# TODO - Do we need to use different kernel params and attestation-agent doesn't get config from
+		# kata-agent anymore based on Ding's PR comments
+		add_kernel_params "agent.aa_kbc_params=cc_kbc::http://${CC_KBS_IP}:${CC_KBS_PORT}/"
```

Comment:
I don't know exactly what's the parameter passed to kernel but I know that we won't use `add_kernel_params` because that function is meant to change the kata's configuration.toml. We will need to intially pass that information via pod annotation.

Perhaps in a near future use agent policy mechanism?

Reply:
Yeah for the time being we still need to set this via kernel param. For platforms that do not support init data we will continue to use this kernel param, but I am hoping we will have an annotation for it (I think Ryan's shim PR adds this for SEV). For other platforms we will move this configuration.

Reply:
Hi @fitzthum ,

ok, my takeaways are:

* KBC is no longer configured through the kata-agent. Corollary: policy agent can't be used to set parameters
* init data is the desired long term solution to set KBS url. But not all platforms support init-data right now. Thus, fallback to kernel parameters

Reply:
Yeah. To be precise the way that the initdata support will probably work is that the kata agent will parse the initdata into multiple configuration files, such as the config for the cdh.

We are removing the `aa_kbc_params` parsing from the kata agent mainly to avoid clutter. The kata agent could still configure the KBC but it seems nicer to have the CDH handle this, especially since there will be a bit of new logic regarding whether there is a cdh config file or whether there is a kernel parameter. We will hide all of that from the kata agent, but the agent will handle initdata if it is supported.

Reply:
so for the time being I will leave `add_kernel_params`